core/src/main/java/org/elasticsearch/cluster/routing/OperationRouting.java

```diff
@@ -55,19 +54,16 @@ public OperationRouting(Settings settings, AwarenessAllocationDecider awarenessA
     }
 
     public ShardIterator indexShards(ClusterState clusterState, String index, String type, String id, @Nullable String routing) {
```

Comment:
@bleskes we introduced the public `shardId` method here, so we can't remove all the changes to this class. I looked into 2.x and I think we can back port it without any issues.